tests/run/test_patma.py

```diff
@@ -0,0 +1,3415 @@
+### COPIED FROM CPython 3.9
+### Original part after ############
+# cython: language_level=3
+
+# new code
+import cython
+from Cython.Compiler.Main import compile as cython_compile, CompileError
+from Cython.Build.Inline import cython_inline
+import contextlib
+from tempfile import NamedTemporaryFile
+
+@contextlib.contextmanager
+def hidden_stderr():
+    try:
+        from StringIO import StringIO
+    except ImportError:
+        from io import StringIO
+
+    old_stderr = sys.stderr
+    try:
+        sys.stderr = StringIO()
+        yield
+    finally:
+        sys.stderr = old_stderr
+
+def _compile(code):
+    with NamedTemporaryFile(suffix='.py') as f:
+        f.write(code.encode('utf8'))
+        f.flush()
+
+        with hidden_stderr():
+            result = cython_compile(f.name, language_level=3)
+    return result
+
+if cython.compiled:
+    def compile(code, name, what):
+        assert what == 'exec'
+        result = _compile(code)
+        if not result.c_file:
+            raise SyntaxError('unexpected EOF')  # compile is only used for testing errors
+
+############## SLIGHTLY MODIFIED ORIGINAL CODE
+import array
+import collections
+import enum
+import inspect
+import sys
+import unittest
+
+if sys.version_info > (3, 10):
+    import dataclasses
+    @dataclasses.dataclass
+    class Point:
+        x: int
+        y: int
+else:
+    # predates dataclasses with match args
+    class Point:
+        __match_args__ = ("x", "y")
+        x: int
+        y: int
+
+# TestCompiler removed - it's very CPython-specific
+# TestTracing also removed - doesn't seem like a core test
+
+# FIXME - return all the "return"s added to cause code to be dropped
+############## ORIGINAL PART FROM CPYTHON
+
+
+class TestInheritance(unittest.TestCase):
+
+    @staticmethod
+    def check_sequence_then_mapping(x):
+        return
```

Comment:
I'd rather see a decorator being used here. It's unhelpful to search for left-over `return`s, but would be easy to search for `@disabled`.

Reply:
I agree but I don't think it'd work. The decorator would remove them at runtime, but `return` drops them before `AnalyseExpressions` (where they just raise the error "Structural pattern match is not yet implemented")

Reply:
Perhaps do `return #  disabled` so it's easy to search for.

Reply:
@TeamSpen210 good suggestion - I'll do that